sample_metadata/parser/cloudhelper.py

```diff
@@ -155,7 +156,12 @@ async def get_gcs_blob(self, filename: str) -> storage.Blob:
 
     def _list_gcs_directory(self, gcs_path) -> list[str]:
 
-        bucket_name, remaining_path = gcs_path[5:].split('/', maxsplit=1)
-        bucket = self.get_gcs_bucket(bucket_name)
-        blobs = self.gcs_client.list_blobs(bucket, prefix=remaining_path, delimiter='/')
-        return [f'gs://{bucket_name}/{blob.name}' for blob in blobs]
+        path = GSPath(gcs_path)
```

Comment:
I'm not really following the change here, is it mostly just the trailing `/` on the path? The only main change I see is the change in the `remaining_path`.

Reply:
The line with `splis` just didn't work:
```
    bucket_name, remaining_path = gcs_path[5:].split('/', maxsplit=1)
ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
I see, cloudpathlib has to be installed for other reasons so this seems reasonable to me